### ISSUE TITLE
Add auto connect mechanism

### ISSUE THREAD

**Issue by chiku524** [aragon/use-wallet#70]
So I am able to connect the wallet successfully, but if I happen to refresh the page, the wallet will disconnect and I need to reconnect it again.

Wouldn't it make sense for the wallet to only disconnect when using the wallet.reset() function?

**Comment by chiku524** [comment]
This is for the 'fortmatic' and 'portis' connectors, by the way. Haven't tested the other ones out yet

**Comment by meirbank** [comment]
Same happens with meta mask. Would be great if we can connect automatically after user connects their wallet for the first time. This should be possible, on sites like Uniswap they do this.

**Comment by vortextemporum** [comment]
Unfortunately I also encountered the same problem. It is the only reason I am not using this component for my project at the moment.

**Comment by bpierre** [comment]
Agreed, it would be a great feature to have! Feel free to open a PR with it, otherwise I’ll have a look after https://github.com/aragon/use-wallet/pull/51 gets merged.

**Comment by stoplion** [comment]
Has anyone found a way around this? I just added useWallet to my app, running into same issue

**Comment by seniorjoinu** [comment]
I just made a couple of functions like

```typescript
export function setWalletConnected(value: keyof Connectors | null) {
  localStorage.setItem('__WALLET_CONNECTED', JSON.stringify(value));
}

export function isWalletConnected(): keyof Connectors | null {
  const val = localStorage.getItem('__WALLET_CONNECTED');

  return val ? JSON.parse(val) : null;
}
```

When a user connects their wallet, I call `setWalletConnected('injected')`. Then, when the app is loaded, I just do this
```typescript
  const connectedId = isWalletConnected();
  const wallet = useWallet();
  if (wallet.status == 'disconnected' && connectedId != null) {
    wallet.connect(connectedId)
  }
```

I don't know if it works for other connectors, but Metamask's one works fine.

**Comment by pioul** [comment]
Another more solid approach is available in the `web3-react` repo: the [`useEagerConnect` hook](https://github.com/NoahZinsmeister/web3-react/blob/bb39007501da7aad4b3f7bd3eab2805c200d0c1b/example/hooks.ts#L6-L31). The code for that hook looks like this once adapted to use-wallet by using the web3-react instance exposed by  `useWallet`:

```js
import { useState, useEffect } from 'react';
import { useWallet } from 'use-wallet';

const useEagerConnect = () => {
  const { _web3ReactContext, connectors } = useWallet();
  const { activate, active } = _web3ReactContext;
  const injected = connectors.injected.web3ReactConnector({ chainId: CHAIN_ID });
  const [tried, setTried] = useState(false);

  useEffect(() => {
    injected.isAuthorized().then((isAuthorized) => {
      if (isAuthorized) {
        activate(injected, undefined, true).catch(() => {
          setTried(true);
        });
      } else {
        setTried(true);
      }
    });
  }, []); // intentionally only running on mount (make sure it's only mounted once :))

  // if the connection worked, wait until we get confirmation of that to flip the flag
  useEffect(() => {
    if (!tried && active) {
      setTried(true);
    }
  }, [tried, active]);

  return tried;
};

export {
  useEagerConnect,
};
```

**Comment by MoukimHF** [comment]
This works fine for me

`  const { account, connect, reset, status } = useWallet();`
`  React.useEffect(() => {
    if(localStorage.getItem("status")==="connected"){
      connect("injected")
    }
  },[]);`

`  const resetCb = () => {
    reset();
    localStorage.setItem("status", "disconnected");
  };`

`  const connectCb = () => {
    connect("injected").then(() => {
      localStorage.setItem("status", "connected");
    })
  }`

**Comment by Morhef** [comment]
.
